core/src/main/java/org/elasticsearch/action/search/SearchScrollQueryAndFetchAsyncAction.java

```diff
@@ -170,7 +171,7 @@ private void finishHim() {
     private void innerFinishHim() throws Exception {
         ScoreDoc[] sortedShardList = searchPhaseController.sortDocs(true, queryFetchResults);
         final InternalSearchResponse internalResponse = searchPhaseController.merge(sortedShardList, queryFetchResults,
-            queryFetchResults);
+                queryFetchResults, null);
```

Comment:
ohhhh I see. Then can you just add javadocs to SearchPhaseController.merge to document that the request will be null for non-search requests such as scrolls?